tests/test_009_layers.py

```diff
@@ -159,6 +158,59 @@ def test_set_minimum_layer_thickness(plot=False):
         plot_test(regis, ds_new)
 
 
+def test_calculate_transmissivity():
+    regis = get_regis_horstermeer()
+    nlmod.layers.calculate_transmissivity(regis)
+
+
+def test_calculate_resistance():
+    regis = get_regis_horstermeer()
+    # with the default value of between_layers=True
+    nlmod.layers.calculate_resistance(regis)
+    # and also with between_layers=False
+    nlmod.layers.calculate_resistance(regis, between_layers=False)
+
+
+def test_get_layer_of_z():
+    regis = get_regis_horstermeer()
+    z = -100
+
+    layer = nlmod.layers.get_layer_of_z(regis, z)
+
+    assert (regis["botm"].isel(layer=layer) < z).all()
+    top = regis["botm"] + nlmod.layers.calculate_thickness(regis)
+    assert (top.isel(layer=layer) > z).all()
+
+
+def test_aggregate_by_weighted_mean_to_ds():
+    regis = get_regis_horstermeer()
+    regis2 = regis.copy(deep=True)
+
+    # botm needs to have the name "bottom'
+    regis2["bottom"] = regis2["botm"]
+    # top needs to be 3d
+    regis2["top"] = regis2["botm"] + nlmod.layers.calculate_thickness(regis2)
+    kh_new = nlmod.layers.aggregate_by_weighted_mean_to_ds(regis, regis2, "kh")
+    assert np.abs(kh_new - regis["kh"]).max() < 1e-5
+    # assert (kh_new.isnull() == regis["kh"].isnull()).all() # does not assert to True...
+
+
+def test_check_elevations_consistency():
+    regis = get_regis_horstermeer()
+    nlmod.layers.check_elevations_consistency(regis)
```

Comment:
perhaps we can add an inconsistent elevation to check method when the layer model isn't correct?

Reply:
I added a check. The inconsistencies are reported by the logger, but it turns out you can use pytest to test logger output as well (at least locally on my computer).